shell/browser/notifications/platform_notification_service.cc

```diff
@@ -119,10 +266,55 @@ void PlatformNotificationService::DisplayPersistentNotification(
     const GURL& service_worker_scope,
     const GURL& origin,
     const blink::PlatformNotificationData& notification_data,
-    const blink::NotificationResources& notification_resources) {}
+    const blink::NotificationResources& notification_resources) {
+  absl::optional<int> proc_id =
+      browser_client_->GetRenderFrameProcessID(service_worker_scope);
+  if (!proc_id.has_value())
+    return;
+
+  content::WebContents* web_ctx =
+      browser_client_->GetWebContentsFromProcessID(proc_id.value());
+  if (!web_ctx)
```

Comment:
Hello @VerteDinde thank you for the remark.

It works fine. Honestly it is too time consuming for me to repeat things from previous discussions. Persistent notifications come from **ServiceWorker** which in it's turn hast last **NotificationContext**. Internaly we store pointer on it and than by function call try to find with usage of pending process list. Process with this context is right place for notification. You can believe me or not, but in case if somebody told me the similar at least I would try to understand developer who made everything almost two years ago. Be sure I had enough time into Debug mode to understand some not obvious things :)

Regards.